### ISSUE TITLE
[BUG] Call is not ended after navigation in Safari 13.1

### ISSUE THREAD

**Issue by ostap0207** [twilio/twilio-client.js#164]
- [x] I have verified that the issue occurs with the latest twilio.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-client.js/blob/master/CHANGELOG.md).
 - [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-client.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [x] I verified that the [Quickstart](https://github.com/TwilioDevEd/client-quickstart-js) application works in my environment.
 - [x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

**Code to reproduce the issue:**
1. Start the client-quickstart-js in Safari 13.1
2. Call your phone number
3. Add a link for navigation in the browser console (it does not matter where the page goes):
```
      var a = document.createElement('a');
      var linkText = document.createTextNode("navigate");
      a.appendChild(linkText);
      a.title = "navigate";
      a.href = "/next_page";
      document.body.appendChild(a);
```
4. Click on the link

**Expected behavior:**

Phone call ends

**Actual behavior:**

Phone call continues after navigation

Debug logs:
```
[Info] Setting up VSP (twilio.min.js, line 99)
[Info] WSTransport.open() called... (twilio.min.js, line 99)
[Info] Attempting to connect... (twilio.min.js, line 99)
[Info] Closing and cleaning up WebSocket... (twilio.min.js, line 99)
[Info] No WebSocket to clean up. (twilio.min.js, line 99)
[Info] WebSocket opened successfully. (twilio.min.js, line 99)
[Info] Setting token and publishing listen (twilio.min.js, line 99)
[Info] Stream is ready (twilio.min.js, line 99)
[Log] Calling ***... (quickstart.js, line 93)
[Info] signalingState is "have-local-offer" (twilio.min.js, line 99)
[Info] pc.iceGatheringState is "gathering" (twilio.min.js, line 99)
[Info] pc.connectionState is "connecting" (twilio.min.js, line 99)
[Info] ICE Candidate: {"candidate":"candidate:*** typ host generation 0 ufrag lREQ network-cost 999","sdpMid":"0","sdpMLineIndex":0} (twilio.min.js, line 99)
[Info] ICE Candidate: null (twilio.min.js, line 99)
[Info] pc.iceGatheringState is "complete" (twilio.min.js, line 99)
[Info] signalingState is "stable" (twilio.min.js, line 99)
[Info] pc.iceConnectionState is "checking" (twilio.min.js, line 99)
[Info] pc.iceConnectionState is "connected" (twilio.min.js, line 99)
[Info] Media connection established. (twilio.min.js, line 99)
[Info] pc.connectionState is "connected" (twilio.min.js, line 99)
[Info] pc.iceConnectionState is "completed" (twilio.min.js, line 99)

WebSocket connection to 'wss://chunderw-vpc-gll.twilio.com/signal' failed: WebSocket is closed due to suspension.
Page navigated at 22:20:29
```

Based on my investigation this could be due to the fact that Safari might cache the page and `unload` event is not called (see https://webkit.org/blog/427/webkit-page-cache-i-the-basics/).

You can also observe this by executing `window.onpagehide = console.log` before navigation and checking the `persisted` property. When persisted is `true` the page was cached and the call is not ended. When it is `false` (for example during page reload) then the page is not cached and the call ends normally.

In Safari 13.1 Changelog (https://developer.apple.com/documentation/safari_release_notes/safari_13_1_release_notes) there is `Improved back-forward responsiveness.` which might have caused this.

**Software versions:**

 - [x] Browser(s): Safari 13.1
 - [x] Operating System: Mac OS 10.15.4
 - [x] twilio.js: 1.10.3
 - [ ] Third-party libraries (e.g., Angular, React, etc.):

**Comment by charliesantos** [comment]
Hi @ostap0207 , thank you for submitting and for the detailed information you provided. Just to clarify, you are seeing this on Safari only and it doesn't happen on other browsers correct?

Thank you,
Charlie

**Comment by ostap0207** [comment]
Yes, we can only see it in Safary 13.1. We haven't seen it in other browsers or Safari 13.0 and below.

**Comment by charliesantos** [comment]
Thank you @ostap0207 . We will investigate.
Jira [CLIENT-7633](https://issues.corp.twilio.com/browse/CLIENT-7633)

**Comment by charliesantos** [comment]
Fixed in 1.12.5
